libs/wire-api/src/Wire/API/Team/Feature.hs

```diff
@@ -257,6 +264,72 @@ instance HasDeprecatedFeatureName 'TeamFeatureDigitalSignatures where
 typeTeamFeatureName :: Doc.DataType
 typeTeamFeatureName = Doc.string . Doc.enum $ cs . toByteString' <$> [(minBound :: TeamFeatureName) ..]
 
+----------------------------------------------------------------------
+-- TeamFeatureTTLValue
+
+-- Using Word to avoid dealing with negative numbers.
+-- Ideally we would also not support zero.
+-- Currently a TTL=0 is ignored on the cassandra side.
+data TeamFeatureTTLValue
+  = TeamFeatureTTLSeconds Word
+  | TeamFeatureTTLUnlimited
+  deriving stock (Eq, Show, Generic)
+
+deriving instance ToJSON TeamFeatureTTLValue
+
+deriving instance FromJSON TeamFeatureTTLValue
+
+instance ToHttpApiData TeamFeatureTTLValue where
+  toQueryParam a = case a of
+    TeamFeatureTTLSeconds d -> T.pack . show $ d
+    TeamFeatureTTLUnlimited -> "unlimited"
+
+instance FromHttpApiData TeamFeatureTTLValue where
+  parseQueryParam = \case
+    "unlimited" -> Right TeamFeatureTTLUnlimited
+    d -> case readEither . T.unpack $ d of
+      Right d' -> Right . TeamFeatureTTLSeconds $ d'
+      Left e -> Left . T.pack $ e <> " oops."
+
+instance S.ToParamSchema TeamFeatureTTLValue where
+  toParamSchema _ = S.toParamSchema (Proxy @Int)
+
+instance ToByteString TeamFeatureTTLValue where
+  builder TeamFeatureTTLUnlimited = "unlimited"
```

Comment:
I sometimes like to implement `FromHttpApiData` in terms of `FromByteString` or vice versa.  Makes for more coherent error handling, for once.  You could also factor out a function that is called in both instances.  But I don't insist.

Reply:
Removing one of them anyway, so no need to 🏌️‍♂️it. :)